tests/Command/UsersCheckPasswordCommandTest.php

```diff
@@ -92,6 +102,31 @@ public function testExecuteFd3($inputStream, $returnCode): void
         $this->assertEquals($returnCode, $commandTester->getStatusCode());
     }
 
+    public function testExecuteCallsLoginListener(): void
+    {
+        $inputStream = "user@example.org\x00password";
+        $manager = $this->getManager();
+        $reader = $this->getReaderFd3($inputStream);
+        $handler = $this->getHandler(true);
+        $mailCryptKeyHandler = $this->getMailCryptKeyHandler();
+        $mailCrypt = 2;
+        $mailUID = 5000;
+        $mailGID = 5000;
+        $mailLocation = 'var/vmail';
+
+        $command = new UsersCheckPasswordCommand($manager,
+            $reader,
+            $handler,
+            $mailCryptKeyHandler,
+            $mailCrypt,
+            $mailUID,
+            $mailGID,
+            $mailLocation);
+        $commandTester = new CommandTester($command);
+
+        $commandTester->execute([]);
```

Comment:
I do not see any assertation here. It would be good to assert that the command returns successfully and the mocked services were called as expected.

Reply:
I changed the code logic now so that `$this->loginListener->expects(self::once())->method('onLogin')` is in the test function itself. Probably it's easier to follow the logic of the test that way.

I also added an assertion to make sure the `execute()` call succeeded, but that's not what we want to test here. We want to test whether `LoginListener->onLogin()` got triggered.

Reply:
I changed the code logic now so that `$this->loginListener->expects(self::once())->method('onLogin')` is in the test function itself. Probably it's easier to follow the logic of the test that way.

I also added an assertion to make sure the `execute()` call succeeded, but that's not what we want to test here. We want to test whether `LoginListener->onLogin()` got triggered.